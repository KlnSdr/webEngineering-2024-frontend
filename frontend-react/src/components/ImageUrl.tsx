import React, { useState, useEffect } from "react";

interface ImageUrlProps {
    url: string;
}

const ImageUrl: React.FC<ImageUrlProps> = ({ url }) => {
    const [isValid, setIsValid] = useState<boolean>(true);

    useEffect(() => {
        const img = new Image();
        img.src = url;
        img.onload = () => setIsValid(true);
        img.onerror = () => setIsValid(false);
    }, [url]);

    return (
        <div className="mt-2">
            {isValid ? (
                <img src={url} alt="dish" style={{ width: "300px", height: "auto" }} />
            ) : (
                <p>Invalid image URL</p>
            )}
        </div>
    );
};
export default ImageUrl;