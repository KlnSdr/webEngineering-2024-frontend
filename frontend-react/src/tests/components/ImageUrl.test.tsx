import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import ImageUrl from "../../components/ImageUrl";

describe("ImageUrl Component", () => {
    test("renders with correct image", () => {
        const origin = "https://via.placeholder.com/150";

        const {asFragment} = render( <ImageUrl url={origin} />);

        const imageElement = screen.getByRole("img");
        expect(imageElement).toHaveAttribute("src", origin);
        expect(imageElement).toHaveAttribute("alt", "dish");
        expect(asFragment()).toMatchSnapshot();
    });
});