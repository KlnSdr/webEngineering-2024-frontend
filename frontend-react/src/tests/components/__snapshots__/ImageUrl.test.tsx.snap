// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUrl Component renders with correct image 1`] = `
<DocumentFragment>
  <div
    class="mt-2"
  >
    <img
      alt="dish"
      src="https://via.placeholder.com/150"
      style="width: 300px; height: auto;"
    />
  </div>
</DocumentFragment>
`;

exports[`ImageUrl Component renders with invalid image 1`] = `
<DocumentFragment>
  <div
    class="mt-2"
  >
    <img
      alt="dish"
      src="https://via.placeholder.com/150"
      style="width: 300px; height: auto;"
    />
  </div>
</DocumentFragment>
`;
